src/App.tsx

```diff
@@ -1,56 +1,72 @@
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList/PeopleList';
+
+function debounce(callback: (arg: string) => void, delay: number) {
+  let timerId = 0;
+
+  return (arg: string) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(arg);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [shownList, setShownList] = useState(false);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => person.name
+      .toLowerCase().includes(appliedQuery.toLowerCase().trim()));
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
+
       </h1>
 
       <div className="dropdown is-active">
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
-            className="input"
+            className="input is-rounded is-primary"
+            value={query}
+            onChange={handleQueryChange}
+            onFocus={() => setShownList(true)}
+            onBlur={() => setShownList(false)}
+
           />
         </div>
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+        {shownList && (
```

Comment:
You need to add "No matching suggestions" if we don`t find any person.
![image](https://github.com/mate-academy/react_autocomplete/assets/63805271/df1ca460-d727-4cbb-b63b-48580944d0ec)

Like this: 
![image](https://github.com/mate-academy/react_autocomplete/assets/63805271/6dfcf882-51bc-4ffa-ae60-cc93d6e3661e)